core/src/main/java/org/axonframework/serialization/CachingSupplier.java

```diff
@@ -0,0 +1,66 @@
+package org.axonframework.serialization;
+
+import java.io.Serializable;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Supplier;
+
+/**
+ * A Supplier that caches its result, useful if the supplier operation is expensive and expected to always yield the
+ * same result. This supplier can be used concurrently by multiple threads.
+ *
+ * @author Rene de Waele
+ */
+public class CachingSupplier<T> implements Supplier<T>, Serializable {
```

Comment:
Why is this class Serializable?
The delegate field is (rightfully) transient. However, there is nothing to ensure that the AtomicReference is filled before the object is serialized.